test/framework/cluster_proxy.go

```diff
@@ -247,20 +265,52 @@ func (p *clusterProxy) GetCache(ctx context.Context) cache.Cache {
 	return p.cache
 }
 
-// Apply wraps `kubectl apply ...` and prints the output so we can see what gets applied to the cluster.
-func (p *clusterProxy) Apply(ctx context.Context, resources []byte, args ...string) error {
-	Expect(ctx).NotTo(BeNil(), "ctx is required for Apply")
-	Expect(resources).NotTo(BeNil(), "resources is required for Apply")
-
-	if err := testexec.KubectlApply(ctx, p.kubeconfigPath, resources, args...); err != nil {
```

Comment:
Let's drop test/framework/exec/kubectl.go & hack/ensure-kubectl.sh (+ it's usages)